### DISCUSSION TITLE
Delete  a specific file from github repository

### DISCUSSION THREAD

**Discussion by harinigautham** [community/community#58962] [Repositories]
### Select Topic Area

Product Feedback

### Body

![image](https://github.com/community/community/assets/102571460/4be366c3-a726-44b7-bc49-17c0bb062e25)

The delete option is unclickable and i want to delete a file from my repository. I don't see the delete bin button as well!
Is this something new with github? I have been able to delete a file before but not now! 
Can anyone please help?

**Comment by DSDmark** [comment]
Hi @harinigautham,

can you share the link of that repo? And some more details about it.

It looks like to relate to permissions issue. Are you the owner of the repository?

**Comment by airtower-luna** [comment]
The screenshot is of a *commit* view, not of the repository at some branch. You can't edit a commit, at least not over the web UI.

If you just want to delete the file with a new commit, view the file at the right branch (instead of in the commit view) and delete it there. The file will remain in the old commits.

If you want to remove the file from the repository history, see [Git Tools - Rewriting History](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History). Note that this will change *all* commit IDs after the first removed or changed commit. This means anyone who already has a copy of your repository and tries to fetch updates will have to manually accept the modified history. :warning: 

In the latter case, the file might still show up in cached data. It it's sensitive and must be *really* gone, [you'll have to contact GitHub support after rewriting the history](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/removing-sensitive-data-from-a-repository) (see the red box near the top).

**Comment by greyesquaxar** [comment]
I have the same problem

**Comment by GabyCT** [comment]
For more information see 
https://docs.github.com/en/repositories/working-with-files/managing-files/deleting-files-in-a-repository

**Reply by AlexanderAdelAU** [reply]
That is not uptodate

**Reply by xTararAisx** [reply]
totally outdated like 85% github doc

**Reply by GeoPotato** [reply]
I followed the instructions in that link, exactly as directed. However, the Delete File option is not there for me. Something has changed.
![image](https://github.com/user-attachments/assets/23052eb4-6dc1-4ce6-96fd-5ec12bf640c3)

EDIT: I figured it out. The Delete File option is OUT OF VIEW in the pop-up window. It is actually a SCROLLING pop-up window. You have to scroll down in the pop-up window for the Delete File option to appear. Why would GitHub design the pop-up window this way?! Here's how it looks after I scroll down:
![image](https://github.com/user-attachments/assets/bccf91aa-45de-499d-8808-d1fe378ed714)

**Reply by arhysp** [reply]
> I followed the instructions in that link, exactly as directed. However, the Delete File option is not there for me. Something has changed. ![image](https://private-user-images.githubusercontent.com/38843257/433569553-23052eb4-6dc1-4ce6-96fd-5ec12bf640c3.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTA2NzE2MDgsIm5iZiI6MTc1MDY3MTMwOCwicGF0aCI6Ii8zODg0MzI1Ny80MzM1Njk1NTMtMjMwNTJlYjQtNmRjMS00Y2U2LTk2ZmQtNWVjMTJiZjY0MGMzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA2MjMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNjIzVDA5MzUwOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWYxMjQ1YWQ5ZmZhMDM4ZDRlN2JlMDc0ZGNlZDExZGI2YjAzYjZkYjA1MjAyYTk4ZGU5ZTIzZWFhNmQwMjNhNmQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.FWYPPmUjTiWl0VbP5nQbEAtmvLBmeAe5yxmpPsNAKxE)
> 
> EDIT: I figured it out. The Delete File option is OUT OF VIEW in the pop-up window. It is actually a SCROLLING pop-up window. You have to scroll down in the pop-up window for the Delete File option to appear. Why would GitHub design the pop-up window this way?! Here's how it looks after I scroll down: ![image](https://private-user-images.githubusercontent.com/38843257/433571010-bccf91aa-45de-499d-8808-d1fe378ed714.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTA2NzE2MDgsIm5iZiI6MTc1MDY3MTMwOCwicGF0aCI6Ii8zODg0MzI1Ny80MzM1NzEwMTAtYmNjZjkxYWEtNDVkZS00OTlkLTg4MDgtZDFmZTM3OGVkNzE0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA2MjMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNjIzVDA5MzUwOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWRmYmI2NmE5ZjA5ZDMwN2I0NzAyYjA3OGYxNjViN2M3MjBhMDA3OTc0ZDY2MDFjNmI2ZmFlMThjNTk3MzA5MzgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.DH9gOOzVsEKXxuL853qqR8epX2Sr8bv46MMs6CUQrdE)

Thank you so much, life saver!

**Comment by Mohsanali-ali** [comment]
i have the same issue
